### DISCUSSION TITLE
.set_ticks(tick_locs) .set_ticklabels

### DISCUSSION THREAD

**Discussion by WWIIWWIIWW** [pyvista/pyvista#1464] [Q&A] [answered]
Hello everyone,
can anyone please assist me in setting up the tick position and ticklabels for colorbar?
In matplotlib, we can have things like (https://stackoverflow.com/questions/42496357/discrete-color-bar-with-tick-labels-in-between-colors):
`fig, ax = plt.subplots(figsize=(12,8))`
`cmap = plt.get_cmap('viridis', n_clusters)`
`scatter = ax.scatter(x, y, c=clusters, cmap=cmap)`
`ax.scatter(clusters_x, clusters_y, c='red')`
`cbar = plt.colorbar(scatter)`
`tick_locs = (np.arange(n_clusters) + 0.5)*(n_clusters-1)/n_clusters`
`cbar.set_ticks(tick_locs)`
`cbar.set_ticklabels(np.arange(n_clusters))`

How about pyvista? I have:
`cmap = colors.ListedColormap(unique_colors)`
`mesh['values'] = cluster_member_colors`
`p.add_mesh(mesh, scalar_bar_args=sargs, scalars='values', cmap=cmap, annotations=annotations)`
Then cmap will show ticklabels same as the scalar values.

How can I set say, -1, 0, 1, 2, 3, 4, ... as the legend label for descrete colormap?

Thanks in advance,
Kai.

**Comment by akaszynski** [comment, marked as answer]
This should work for you:
```py
import pyvista

mesh = pyvista.Sphere()
scalars = mesh.points[:, 2]*10

sargs = {'fmt': '%.0f', 'n_labels': 5, 'title': 'my cool scalars', 'color': 'k'}
pl = pyvista.Plotter()
pl.add_mesh(mesh, scalars=scalars, scalar_bar_args=sargs)
pl.background_color = 'w'
pl.show()
```

![sc](https://user-images.githubusercontent.com/11981631/123821383-94628a00-d8b8-11eb-99c3-49ca3afa6c38.png)

**Reply by WWIIWWIIWW** [reply]
Hi,
Thanks for your reply. It is not exactly what I am looking for, while indeed it provides some hints on how to solve the issue

Bests,
Kai.

**Comment by WWIIWWIIWW** [comment]
![Screenshot from 2021-06-30 12-37-41](https://user-images.githubusercontent.com/45393386/123946904-01048400-d9a0-11eb-9f64-bec4bd6d0ade.png)
One more way I am trying...Instead of using the default colorbar, I chose to use legend as seen in the picture.
The problem is how to change the triangle to some other shapes? e.g. a circle or a rectangle with the right side numbers written within it?

```py
legend_elements = [[str(unique_label), cmap.colors[i]] for i, unique_label in enumerate(unique_labels)]
'p.add_legend(legend_elements)
```

Kai.

**Reply by akaszynski** [reply]
Not really and that's something we should add.  Can you please open up a new issue as a feature request?  This is something we're lacking.

**Reply by WWIIWWIIWW** [reply]
done. While not sure how to label the issue as feature request :D

**Reply by akaszynski** [reply]
Just saw it.  We'll try to work on it soon.
